client/scripts/views/pages/create_community/eth_dao_form.ts

```diff
@@ -0,0 +1,238 @@
+import 'pages/create_community.scss';
+
+import m from 'mithril';
+import app from 'state';
+import { Table, Button } from 'construct-ui';
+import $ from 'jquery';
+import { initAppState } from 'app';
+import { slugify } from 'utils';
+import { ChainBase, ChainNetwork, ChainType } from 'types';
+import { isAddress } from 'web3-utils';
+import Web3 from 'web3';
+import { IAaveGovernanceV2__factory } from 'eth/types';
+import { notifyError } from 'controllers/app/notifications';
+import {
+  InputPropertyRow, SelectPropertyRow
+} from 'views/components/metadata_rows';
+import CompoundAPI, { GovernorTokenType, GovernorType } from 'controllers/chain/ethereum/compound/api';
+import AaveApi from 'controllers/chain/ethereum/aave/api';
+import { initChainForm, defaultChainRows, EthChainAttrs, EthFormState, ethChainRows } from './chain_input_rows';
+
+type EthDaoFormAttrs = EthChainAttrs;
+
+interface EthDaoFormState extends EthFormState {
+  token_name: string;
+  id: string;
+  name: string;
+  symbol: string;
+  network: ChainNetwork.Aave | ChainNetwork.Compound,
+  saving: boolean;
+  loading: boolean;
+  status: string;
+  error: string;
+}
+
+const EthDaoForm: m.Component<EthDaoFormAttrs, EthDaoFormState> = {
+  oninit: (vnode) => {
+    vnode.state.chain_string = 'Ethereum Mainnet';
+    vnode.state.chain_id = '1';
+    vnode.state.url = vnode.attrs.ethChains[1];
+    vnode.state.address = '';
+    vnode.state.token_name = 'token';
+    vnode.state.id = '';
+    vnode.state.name = '';
+    vnode.state.symbol = '';
+    vnode.state.network = ChainNetwork.Compound;
+    initChainForm(vnode.state);
+    vnode.state.saving = false;
+    vnode.state.loaded = false;
+    vnode.state.loading = false;
+    vnode.state.status = '';
+    vnode.state.error = '';
+  },
+  view: (vnode) => {
+    const validAddress = isAddress(vnode.state.address);
+    const disableField = !validAddress || !vnode.state.loaded;
+
+    const updateDAO = async () => {
+      if (!vnode.state.address || !vnode.state.chain_id || !vnode.state.url) return;
+      vnode.state.loading = true;
+      vnode.state.status = '';
+      vnode.state.error = '';
+      try {
+        if (vnode.state.network === ChainNetwork.Compound) {
+          const provider = new Web3.providers.WebsocketProvider(vnode.state.url);
+          const compoundApi = new CompoundAPI(
+            null,
+            vnode.state.address,
+            provider,
+          );
+          await compoundApi.init(vnode.state.token_name);
+          if (!compoundApi.Token) {
+            throw new Error('Could not find governance token. Is "Token Name" field valid?');
```

Comment:
we should also suggest to the user "Try all lowercase if you're sure on spelling" or something similar. I was testing with COMP and it needed to be 'comp'

Reply:
I will add a "case sensitive" note.